### PR TITLE
Add kiali scenarios to graph node from VS spec

### PR DESCRIPTION
**Pull request by nrfox**
Updates the istio appender to add the kiali scenario node metadata labels based on the VS spec instead of the presence of the wizard label.

Fixes: #4090

### CODE REVIEW

File: models/virtual_service.go [hunk 1]
```diff
@@ -67,3 +67,129 @@ func (vService *VirtualService) IsValidHost(namespace string, serviceName string
...
+
+	if routes, isSlice := vService.Spec.Http.([]interface{}); isSlice {
+		for _, route := range routes {
+			if routeMap, isMap := route.(map[string]interface{}); isMap {
+				if _, hasTimeout := routeMap["timeout"]; hasTimeout {
+					return true
+				}
+			}
+		}
+	}
```
Comment (nrfox) [inline models/virtual_service.go:85]: imo ideally the `VirtualService` model would be fully typed and then these methods just performing type checking would no longer be necessary. I'm happy to make that change and remove these methods. Could Kiali simply import and use the upstream istio api types e.g. something like `networkingv1beta1.VirtualService`?

File: graph/types.go [hunk 2]
```diff
@@ -8,24 +8,18 @@ import (
...
-	NodeTypeUnknown               string = "unknown" // The special "unknown" traffic gen node
-	NodeTypeWorkload              string = "workload"
-	TF                            string = "2006-01-02 15:04:05" // TF is the TimeFormat for timestamps
-	Unknown                       string = "unknown"             // Istio unknown label value
-	WizardFaultInjectionLabel     string = "fault_injection"
-	WizardLabelKey                string = "kiali_wizard" // Label the front-end wizards add to objects created by them
-	WizardRequestRoutingLabel     string = "request_routing"
-	WizardRequestTimeoutsLabel    string = "request_timeouts"
-	WizardTCPTrafficShiftingLabel string = "tcp_traffic_shifting"
-	WizardTrafficShiftingLabel    string = "traffic_shifting"
```
Comment (jshaughn) [inline graph/types.go:28]: Easy come, easy go :-)

File: models/virtual_service.go (hunk 1 above)
Comment (jshaughn) [reply models/virtual_service.go:85]: This is probably a better question for @lucasponce .  I think perhaps the models are parsed out the way they in an effort to simplify the structure to something more easy for us to deal with at higher levels.  Perhaps this badging stuff could be parsed out immediately, or stored lazily on the model, to possibly avoid doing the work many times.

Comment (jshaughn) [reply models/virtual_service.go:85]: @nrfox Now that @lucasponce is on PTO, I think we should leave this alone until he comes back and can comment.

Comment (jshaughn) [reply models/virtual_service.go:85]: I will test this as-is.

**Review by jshaughn — APPROVED** [review]
Looks good to me.  The model impl can be discussed later with @lucasponce, and perhaps we could look at parsing out some of this info directly into the model, I'm not sure.  But this seems to be working fine as-is.

image

**Review by pbajjuri20 — APPROVED** [review]
Tested and verified  on Kiali UI 1.38.0 (156692a97f02b2d82716e2f78deb49913fdd9598)


kiali_wizard

**Pull request merged**